### PR TITLE
InstcountCI: Changes how code size is calculated 

### PR DESCRIPTION
**Pull request by Sonicadvance1**
Due to how jit block tail padding is working, there's no real good way
to determine the true "implementation size" of an instruction without
the backend being aware of wanting to investigate it.

Trying to inject another instruction, or another IR operation actually
subtly changes codegen in a way that gives invalid results. The only
real way to get around this is to inject a known token in to the
instruction stream as we `ExitFunction`.

So inject a `udf #0x420f`, and change the scanning behaviour to find the
first one and cut everything else off afterwards.

This already scoops out some code in some game blocks that were
accidentally landing ExitFunction code in the json.

This also has been tested to work with https://github.com/FEX-Emu/FEX/pull/4528 with its InstCountCI
specific changes reverted.

This means we don't need to play subtle padding tricks in the JIT to get
the information we want in InstcountCI.

### CODE REVIEW

File: Source/Tools/CodeSizeValidation/Main.cpp [hunk 1]
```diff
@@ -144,12 +131,18 @@ bool CodeSizeValidation::ParseMessage(const char* Message) {
...
+      CurrentStats.second.erase(CurrentStats.second.begin(), CurrentStats.second.begin() + HeaderSize);
     }
-    if (BaseStats.TailSize) {
-      CurrentStats.second.erase(CurrentStats.second.end() - BaseStats.TailSize, CurrentStats.second.end());
+    // Find the first `udf #0x420f` and remove everything from that point onward.
+    for (auto it = CurrentStats.second.begin(); it != CurrentStats.second.end(); ++it) {
+      if (*it == "udf #0x420f") {
+        CurrentStats.second.erase(it, CurrentStats.second.end());
+        break;
+      }
```
Comment (neobrain) [inline Source/Tools/CodeSizeValidation/Main.cpp:142]:
```suggestion
    auto EraseBegin = std::find(CurrentStats.second.begin(), CurrentStats.second.end(), "udf #420f");
    CurrentStats.second.erase(EraseBegin, CurrentStats.second.end());
```

File: FEXCore/Source/Interface/Core/JIT/BranchOps.cpp [hunk 2]
```diff
@@ -49,6 +49,10 @@ DEF_OP(ExitFunction) {
 
   ResetStack();
 
+  if (CTX->HostFeatures.IsInstcountCI) [[unlikely]] {
+    udf(0x420F);
```
Comment (neobrain) [inline FEXCore/Source/Interface/Core/JIT/BranchOps.cpp:53]:
```suggestion
    // Emit function end marker
    udf(0x420F);
```

File: FEXCore/include/FEXCore/Core/HostFeatures.h [hunk 3]
```diff
@@ -43,6 +43,9 @@ struct HostFeatures {
   bool SupportsAFP {};
   bool SupportsFloatExceptions {};
 
+  // Flag if this is InstCountCI
+  bool IsInstcountCI {};
```
Comment (neobrain) [inline FEXCore/include/FEXCore/Core/HostFeatures.h:47]: Nit:
```suggestion
  bool IsInstCountCI {};
```

File: FEXCore/unittests/Emitter/System_Tests.cpp [hunk 4]
```diff
@@ -6,6 +6,9 @@
 
 using namespace ARMEmitter;
 
+TEST_CASE_METHOD(TestDisassembler, "Emitter: Reserved") {
+  TEST_SINGLE(udf(0), "udf #0x0");
```
Comment (neobrain) [inline FEXCore/unittests/Emitter/System_Tests.cpp:10]: Does it make sense to add a test case for the maximum reserved value as well?

**Review by neobrain — APPROVED** [review]
I'll take this any day over the workaround in my PR - thanks!

File: FEXCore/unittests/Emitter/System_Tests.cpp (hunk 4 above)
Comment (Sonicadvance1) [reply FEXCore/unittests/Emitter/System_Tests.cpp:10]: Sure, added one.
